### ISSUE TITLE
Color keyword argument - Value error

### ISSUE THREAD

**Issue by Seeth92** [parrt/dtreeviz#288]
I am working on a binary classification problem using lightGBM. 
The model was trained on 42 features. The training dataset size is (78000, 42) - 78000 observations spanning across 42 features
The test dataset size is (25220, 42) 

Using dtreeviz on my trained model:

`viz = dtreeviz.model(gm, tree_index = 0, X_train = X_train, y_train=Y_train, feature_names = features, target_name="A", class_names = ["A", "B"])`

When I execute `viz.view()`  I am facing the following error:
`ValueError: The 'color' keyword argument must have one color per dataset, but 1 datasets and 0 colors were provided`

Any thoughts on how to go about this?

**Comment by tlapusan** [comment]
Is it something similar with https://github.com/parrt/dtreeviz/issues/280?

**Comment by baligoyem** [comment]
I am facing the same error. 

Here is the image which contains some details:

![image](https://user-images.githubusercontent.com/61024126/233940640-defc74cb-dd5c-41fc-8b0a-c36d9ebcaffa.png)

**Comment by Seeth92** [comment]
@tlapusan Yes, the error description is the same as the one posted by @baligoyem

**Comment by baligoyem** [comment]
Have you ever faced the AttributeError, which its description is 'Rectangle' object has no attribute 'patches'?

I am asking this question because I have sometimes randomly faced these two errors that are related to each other, I believe.

**Comment by tlapusan** [comment]
did you try the latest version of dtreeviz ?

**Comment by Seeth92** [comment]
Using colour-0.1.5  and  dtreeviz-2.2.1 .. No luck at all

**Comment by tlapusan** [comment]
could you provide a google collab or any kind of shareable notebook so I could reproduce your issue?

**Comment by Seeth92** [comment]
@tlapusan Sorry for responding this late. Unfortunately, I can't share the notebook as the data and features used is confidential :(

**Comment by Guido-Hwang** [comment]
+1

**Comment by Thegongyx** [comment]
+1

**Comment by leonswl** [comment]
+1

**Comment by andylokandy** [comment]
+1

**Comment by windyd** [comment]
In my case (dtreeviz=2.2.2), it seems to be a precision problem from the `get_thresholds` method. If you have small float thresholds, samples are assigned to wrong paths. In some cases, some nodes may end up with no samples. 

```python
class ShadowLightGBMTree(ShadowDecTree):
    ...
    def get_thresholds(self) -> np.ndarray:
        if self.thresholds is not None:
            return self.thresholds

        node_thresholds = [-1] * self.nnodes()
        for i in range(self.nnodes()):
            if self.children_left[i] != -1 and self.children_right[i] != -1:
                if self.is_categorical_split(i):
                    node_thresholds[i] = list(map(int, self.tree_nodes[i]["threshold"].split("||")))
                else:
                    ###  thresholds are ROUNDED!
                    node_thresholds[i] = round(self.tree_nodes[i]["threshold"], 2)

        self.thresholds = np.array(node_thresholds, dtype=object)
        return self.thresholds
```

No sample -> No color mapped -> this problem.

**Comment by leocwolter** [comment]
+1, any workarounds?

**Comment by CaoHaiNam** [comment]
I dealt with this issue. You should ensure that the data you use to train the LGBM model is the same as the data for visualization.

**Comment by sidml** [comment]
I am using the same data for visualization and I still get the error. Not sure, how to fix it.

**Comment by gitgud5000** [comment]
+1

**Comment by worthy7** [comment]
I think this might be to do with having a model trained using a pipe first, having `StandardScaler` in there would mean the data trained into the model is different from what is provided to dtreeviz
